app/controllers/webhooks/base_controller.rb

```diff
@@ -0,0 +1,49 @@
+# This is the base controller of Exercism's webhooks.
+#
+# It is intended to only be called from GitHub webhooks
+# and should not be called by any other source
+
+module Webhooks
+  class BaseController < ApplicationController
+    skip_before_action :verify_authenticity_token
+    skip_before_action :authenticate_user!
+
+    before_action :verify_github_webhook
+
+    layout false
+
+    private
+    def verify_github_webhook
+      return if signature_valid?
+
+      render json: {
+        error: {
+          type: :invalid_webhook_signature,
+          message: "The auth token provided is invalid. Delivery: #{github_delivery}"
+        }
+      }, status: :internal_server_error
```

Comment:
The API controllers use I8N for the error messages. I could do that here, but it didn't know how to add in the delivery argument (which I've copied from the v2 code).

Reply:
I don't think we need i18n here as it's not public facing.